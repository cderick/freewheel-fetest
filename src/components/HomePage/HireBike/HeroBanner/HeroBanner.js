import React from 'react'
import PropTypes from 'prop-types'
import s from './HeroBanner.module.scss'

class HeroBanner extends React.Component {
	constructor(props) {
		super(props)
		this.handleChange = this.handleChange.bind(this)
	}

	handleChange(e){
		if(e){
			const value = e.target.value
			if(value.length > 1){
				this.props.getApiQuery(value)
			} else {
				this.props.cleanData()
			}
		}
	}

	render() {
		if (!this.props.heroBanner) return null
		const { heroMessage, heroForms } = this.props.heroBanner
		return (
			<div id="homePage" className={`container-fluid ${s.containerBackground}`}>
				<div className="row h-100">
					<div className="col my-auto">
						<h1 className="display-4">{heroMessage && heroMessage.title ? heroMessage.title : ''}</h1>
						<form>
							<div className="row">
								<div className="col-10 m-auto">
									{heroForms && heroForms.length
										&& heroForms.map((cv, ind) => (
											<div key={`main${ind}`} className={`form-group text-left ${s.forms} ${cv.customClass ? s[cv.customClass] : ''}`}>
												<label className="text-white" htmlFor={cv.id}>{cv.label}</label>
												<input type="text" onChange={(e) => {this.handleChange(e)}} className="form-control" id={cv.id} placeholder={cv.placeholder} />
											</div>
										)
									)}
								</div>
							</div>
						</form>
					</div>
				</div>
			</div>
		)
	}
}

HeroBanner.propTypes = {
	heroBanner: PropTypes.objectOf(PropTypes.any),
	getApiQuery: PropTypes.func,
	cleanData: PropTypes.func
}

HeroBanner.defaultProps = {
	heroBanner: null,
}

export default HeroBanner
