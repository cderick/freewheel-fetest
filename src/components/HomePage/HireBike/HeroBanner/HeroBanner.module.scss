.containerBackground {
  background:
    linear-gradient(
      rgba(0, 0, 0, 0.6),
      rgba(0, 0, 0, 0.6)
    ),
    url('../../../../static/cyclebanner.png');
  background-position: center center;
  background-repeat: no-repeat;
  background-size: cover;
  padding: 20px 0 70px;

  .customMargin {
    margin-top: 31px;
  }

  .forms {
    width: 33%;
    float: left;
    margin-right: 25px;
  }

  h1 {
    padding: 40px 0 20px 120px;
    font-weight: bolder;
    font-size: 4.3rem;
    letter-spacing: 2px;
    color: white;
  }
}
