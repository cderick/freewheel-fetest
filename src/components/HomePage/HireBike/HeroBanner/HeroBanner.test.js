import React from 'react'
import renderer from 'react-test-renderer'
import HeroBanner from './HeroBanner'

const minProps = {
   heroBanner: {
      heroMessage: {
          title: 'Cycle hire scheme.'
      },
      heroForms: [{
          label: 'Find a cycle to hire:',
          id: 'location',
          placeholder: 'London, SW20, Waterloo...',
      }]
   }
}

const componentNoprops = renderer.create(<HeroBanner />)
const componentWithProps = renderer.create(<HeroBanner {...minProps}  />)

describe('Test on <HeroBanner /> component', () => {
   test('should not render with no props', () => {
      const treeNoprops = componentNoprops.toJSON()
      expect(treeNoprops).toMatchSnapshot()
   })
   test('should render with min props', () => {
      const treeProps = componentWithProps.toJSON()
      expect(treeProps).toMatchSnapshot()
   })
})