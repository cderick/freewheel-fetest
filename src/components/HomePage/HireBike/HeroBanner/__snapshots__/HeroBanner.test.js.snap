// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <HeroBanner /> component should not render with no props 1`] = `null`;

exports[`Test on <HeroBanner /> component should render with min props 1`] = `
<div
  className="container-fluid containerBackground"
  id="homePage"
>
  <div
    className="row h-100"
  >
    <div
      className="col my-auto"
    >
      <h1
        className="display-4"
      >
        Cycle hire scheme.
      </h1>
      <form>
        <div
          className="row"
        >
          <div
            className="col-10 m-auto"
          >
            <div
              className="form-group text-left forms "
            >
              <label
                className="text-white"
                htmlFor="location"
              >
                Find a cycle to hire:
              </label>
              <input
                className="form-control"
                id="location"
                onChange={[Function]}
                placeholder="London, SW20, Waterloo..."
                type="text"
              />
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
